File: python/classes/class_static_function_access.py
#!/usr/bin/python3

class Token:
    number = 0

    def __init__(self):
        Token.number = 0
        self.color = ""

    def update(self, number, color):
        Token.number = number
        self.color = color

    def reset():
        Token.number = 0

Token.reset()
print(f"Token.number: {Token.number}")

print()

some_token = Token()
some_token.update(5, "magenta")

print(f"some_token.number: {some_token.number}, some_token.color: {some_token.color}")

print()

File: ruby/classes/class_static_function_access.rb
#!/usr/bin/ruby2.7

class Token
  @@number = 0
  attr_accessor :color;
  def self.number
    @@number
  end
  def self.number=number
    @@number = number
  end

  def update(number, color)
    @@number = number
    @color = color
  end
  def self.reset()
    @@number = 0
  end
end

Token.reset()
puts("Token.number: #{Token.number}")

puts()

some_token = Token.new

some_token.update(5, "magenta")
puts("some_token.color: #{some_token.color}")
puts("Token.number: #{Token.number}")

puts()
